front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -14,10 +14,75 @@ import type {
   SimulationResponseSuccess,
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
-import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import { mmToKm, mmToM, msToKmh, mToKm } from 'utils/physics';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagInput = {
+  boundaries: number[];
+  values: {
+    speed: number;
+    source: { speed_limit_source_type: string; tag?: string } | null;
+  }[];
+};
+
+const getTagName = (source: { speed_limit_source_type: string; tag?: string } | null): string => {
+  if (source === null) {
+    return 'missing_from_train';
+  }
+  switch (source.speed_limit_source_type) {
+    case 'given_train_tag':
+      return source.tag || '';
+    case 'fallback_tag':
+      return source.tag || '';
+    case 'unknown_tag':
+      return 'incompatible';
+    case 'UU':
+      return source.tag || '';
+    default:
+      return 'missing_from_train'; // Fallback for any unknown types
+  }
+};
+
+const getTagColor = (sourceType: string): string => {
```

Comment:
- allow undefined param
- l79, you can remove the `| ''`, because now undefined is handled
```
const getTagColor = (sourceType?: string): string => {
```